aptos-move/framework/aptos-framework/sources/configs/Stake.move

```diff
@@ -98,124 +108,275 @@ module AptosFramework::Stake {
-        let addr = Signer::address_of(account);
-        let stake_pool = borrow_global_mut<StakePool>(from);
-        let delegation = withdraw_internal(&mut stake_pool.inactive, addr);
-        let Delegation {coins, rewards, locked_until_secs: _} = delegation;
-        Coin::merge<TestCoin>(&mut coins, rewards);
-        coins
+    /// This is only called during Genesis, which is where MintCapability<TestCoin> can be created.
+    /// Beyond genesis, no one can create TestCoin mint/burn capabilities.
+    public(friend) fun store_test_coin_mint_cap(core_resources: &signer, mint_cap: MintCapability<TestCoin>) {
+        move_to(core_resources, TestCoinCapabilities { mint_cap })
     }
 
-    /// Initialize the ValidatorInfo for account.
+    /// Initialize the validator account and give ownership to the signing account.
     public(script) fun register_validator_candidate(
         account: &signer,
         consensus_pubkey: vector<u8>,
         network_address: vector<u8>,
-        fullnode_address: vector<u8>
+        fullnode_address: vector<u8>,
     ) {
+        let account_address = Signer::address_of(account);
+        assert!(!exists<StakePool>(account_address), EALREADY_REGISTERED);
+
         move_to(account, StakePool {
-            voting_power: 0,
-            next_epoch_voting_power: 0,
-            active: IterableTable::new(),
-            pending_active: IterableTable::new(),
-            pending_inactive: IterableTable::new(),
-            inactive: IterableTable::new(),
+            active: Coin::zero<TestCoin>(),
+            pending_active: Coin::zero<TestCoin>(),
+            pending_inactive: Coin::zero<TestCoin>(),
+            inactive: Coin::zero<TestCoin>(),
+            locked_until_secs: 0,
+            operator_address: account_address,
         });
         move_to(account, ValidatorConfig {
             consensus_pubkey,
             network_address,
             fullnode_address,
         });
+        move_to(account, OwnerCapability {
+           pool_address: account_address,
+        });
+    }
+
+    /// Extract and return owner capability from the signing account.
+    public(script) fun extract_owner_cap(account: &signer): OwnerCapability acquires OwnerCapability {
+        move_from<OwnerCapability>(Signer::address_of(account))
+    }
+
+    /// Deposit `owner_cap` into `account`. This requires `account` to not already have owernship of another
+    /// staking pool.
+    public(script) fun deposit_owner_cap(account: &signer, owner_cap: OwnerCapability) {
+        move_to(account, owner_cap);
+    }
+
+    /// Allows an owner to change the operator of the stake pool.
+    public(script) fun set_operator(account: &signer, new_operator: address) acquires OwnerCapability, StakePool {
+        let account_addr = Signer::address_of(account);
+        let ownership_cap = borrow_global<OwnerCapability>(account_addr);
+        set_operator_with_cap(account_addr, ownership_cap, new_operator);
+    }
```

Comment:
This functionality should be discussed separately. I'm not convinced of its utility. There's probably better ways to achieve this without making core framework changes.

Reply:
I'm thinking of this in the context of separating node operator and funds owner (i.e. delegator). If the node operator turns malicious, there needs to be a way for the owner to remove them. This is necessary for both centralized and decentralized staking services

Reply:
I understand. I think we should evaluate delegation independently before we add this. I think we can make delegation work, so I'd rather not introduce a crutch just yet.